src/util/proxyDevices.tsx

```diff
@@ -0,0 +1,146 @@
+import { Input, Label, Select } from "@canonical/react-components";
+import { MainTableRow } from "@canonical/react-components/dist/components/MainTable/MainTable";
+import { getConfigurationRowBase } from "components/ConfigurationRow";
+import { LxdProxyDevice } from "types/device";
+import { ensureEditMode } from "./instanceEdit";
+import { proxyAddressTypeOptions } from "./instanceOptions";
+import { InstanceAndProfileFormikProps } from "components/forms/instanceAndProfileFormValues";
+
+type ConnectionType = "listen" | "connect";
+
+export const getProxyAddress = (
+  customRows: MainTableRow[],
+  device: LxdProxyDevice,
+  index: number,
+  connectionType: ConnectionType,
+  formik: InstanceAndProfileFormikProps,
+  headingTitle: string,
+) => {
+  const deviceParts = device[connectionType]?.split(":") || [];
+  const deviceType = deviceParts.length > 0 ? deviceParts[0] : "tcp";
+  const deviceAddress = deviceParts.length > 1 ? deviceParts[1] : "";
+  const devicePort = deviceParts.length > 2 ? deviceParts[2] : "";
+
+  customRows.push(
+    getConfigurationRowBase({
+      className: "no-border-top inherited-with-form p-heading--6",
+      configuration: headingTitle,
+      inherited: <></>,
+      override: "",
+    }),
+  );
+
+  customRows.push(
+    getConfigurationRowBase({
+      className: "no-border-top inherited-with-form",
+      configuration: (
+        <Label forId={`devices.${index}.${connectionType}`}>Type</Label>
+      ),
+      inherited: (
+        <Select
+          id={`devices.${index}.${connectionType}`}
+          onChange={(e) => {
+            ensureEditMode(formik);
+            const newType = e.target.value;
+            void formik.setFieldValue(
+              `devices.${index}.${connectionType}`,
+              `${newType}:${deviceAddress}:${devicePort}`,
+            );
+
+            if (device.nat === "true") {
+              void formik.setFieldValue(
+                `devices.${index}.connect`,
+                `${newType}:${deviceAddress}:${devicePort}`,
```

Comment:
We should not override the deviceAddress and devicePort of the connect. with this we are taking all the values from listen and copy them to connect, but we should only copy the type.

Reply:
So we need to read the `device.index.connect` value and decompose it to get the address and port to use it here inside the conditional block.